components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementAdminService.java

```diff
@@ -119,17 +119,29 @@ public ServiceProvider getApplication(String applicationName) throws IdentityApp
      */
     public ApplicationBasicInfo[] getAllApplicationBasicInfo() throws IdentityApplicationManagementException {
 
+        return getApplicationBasicInfo("*");
+    }
+
+    /**
+     * Get all basic application information for a matching filter.
+     *
+     * @param filter Application name filter
+     * @return Application Basic Information array
+     * @throws org.wso2.carbon.identity.application.common.IdentityApplicationManagementException
+     */
+    public ApplicationBasicInfo[] getApplicationBasicInfo(String filter)
+            throws IdentityApplicationManagementException {
+
         try {
             applicationMgtService = ApplicationManagementService.getInstance();
-            ApplicationBasicInfo[] applicationBasicInfos =
-                    applicationMgtService.getAllApplicationBasicInfo(getTenantDomain(), getUsername());
+            ApplicationBasicInfo[] applicationBasicInfos = applicationMgtService.
+                    getApplicationBasicInfo(getTenantDomain(), getUsername(), filter);
             ArrayList<ApplicationBasicInfo> appInfo = new ArrayList<>();
             for (ApplicationBasicInfo applicationBasicInfo : applicationBasicInfos) {
                 if (ApplicationMgtUtil.isUserAuthorized(applicationBasicInfo.getApplicationName(), getUsername())) {
                     appInfo.add(applicationBasicInfo);
                     if (log.isDebugEnabled()) {
-                        log.debug("Retrieving basic information of application: " +
-                                applicationBasicInfo.getApplicationName());
+                        log.debug("Application Name:" + applicationBasicInfo.getApplicationName());
```

Comment:
Shall we keep the original debug log "Retrieving basic information of application" as it has more context?